control.go

```diff
@@ -453,7 +453,11 @@ func (c *controlConn) query(statement string, values ...interface{}) (iter *Iter
 			Logger.Printf("control: error executing %q: %v\n", statement, iter.err)
 		}
 
-		q.attempts++
+		host := c.getConn().host
```

Comment:
Collapse this down to:
```go
c.session.mu.Lock()
q.getHostMetrics(c.getConn().host).attempts++
c.session.mu.Unlock()
```

It's just a lot of variables for no purpose. You can leave the `host :=` line but I'd definitely collapse the attempt increase because it reads better.

Reply:
I'm not sure this one follows the regular query execution. It seems we're constructing and executing the statement directly on conn, and not via the reqular query mechanisms. The retry policy is not constructed, which I reckon means it would be set to `SimpleRetryPolicy{numAtt: 3}`...so we still need to count attempts, but probably not locking.... :-)